device.c

```diff
@@ -316,25 +324,42 @@ int adev_open_output_stream(struct audio_hw_device *dev,
         return -EINVAL;
     }
     /* check requested device */
-    /* TODO we should handle only devices described in XML configuration and "default" one */
     if ((devices & AUDIO_DEVICE_BIT_IN) != 0) {
         ALOGE("Failed. Incorrect device type. -EINVAL");
         *stream_out = NULL;
         pthread_mutex_unlock(&xdev->lock);
         return -EINVAL;
     }
 
-    /* TODO define card_id and dev_id for stream, using provided parameters:
-       devices, flags, address, source */
+    /* For now we have one card, so card id is always constant */
+    /* Get device id. We expect (same as parser in car audio service)
+     * that address has format "busX_%s" */
+    if (sscanf(address, "bus%u", &dev_id) != 1) {
+        ALOGE("Address has not supported format.");
```

Comment:
I would change wording here to be more... precise?

Reply:
Done. Explanation was added to message.